src/Fixer/AttributeNotation/OrderedAttributesFixer.php

```diff
@@ -0,0 +1,349 @@
+
+/**
+ * @author HypeMC <hypemc@gmail.com>
+ */
+final class OrderedAttributesFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    public const ORDER_ALPHA = 'alpha';
+    public const ORDER_NONE = 'none';
+
+    private const SUPPORTED_SORT_ALGORITHMS = [
+        self::ORDER_ALPHA,
+        self::ORDER_NONE,
+    ];
+
+    private ?NamespaceAnalysis $namespaceAnalysis = null;
+
+    /**
+     * @var null|array<string, NamespaceUseAnalysis>
+     */
+    private ?array $namespaceUseAnalyses = null;
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Sorts attributes using the configured sort algorithm.',
+            [
+                new CodeSample(
+                    <<<'EOL'
+                        <?php
+
+                        #[Foo]
+                        #[Bar(3)]
+                        #[Qux(new Bar(5))]
+                        #[Corge(a: 'test')]
+                        class Sample1 {}
+
+                        #[
+                            Foo,
+                            Bar(3),
+                            Qux(new Bar(5)),
+                            Corge(a: 'test'),
+                        ]
+                        class Sample2 {}
+
+                        EOL,
+                ),
+                new CodeSample(
+                    <<<'EOL'
+                        <?php
+
+                        use A\B\Foo;
+                        use A\B\Bar as BarAlias;
+                        use A\B as AB;
+
+                        #[Foo]
+                        #[BarAlias(3)]
+                        #[AB\Qux(new Bar(5))]
+                        #[\A\B\Corge(a: 'test')]
+                        class Sample1 {}
+
+                        EOL,
+                    ['sort_algorithm' => self::ORDER_NONE, 'order' => ['A\\B\\Qux', 'A\\B\\Bar', 'A\\B\\Corge']],
+                ),
+            ],
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return \defined('T_ATTRIBUTE') && $tokens->isTokenKindFound(T_ATTRIBUTE);
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('sort_algorithm', 'How the attributes should be sorted.'))
+                ->setAllowedValues(self::SUPPORTED_SORT_ALGORITHMS)
+                ->setDefault(self::ORDER_ALPHA)
+                ->getOption(),
+            (new FixerOptionBuilder('order', 'A list of FQCNs of attributes defining the desired order used when no sorting algorithm is configured.'))
+                ->setAllowedTypes(['string[]'])
+                ->setDefault([])
+                ->setNormalizer(static fn (Options $options, array $value): array => array_flip(array_values(
+                    array_map(static fn (string $attribute): string => ltrim($attribute, '\\'), $value),
+                )))
```

Comment:
I understand that flip is here in order to use attributes' map for sorting purposes, but consider this:

```shell
php > var_dump(array_flip(['a', 'b', 'a']));
array(2) {
  ["a"]=>
  int(2)
  ["b"]=>
  int(1)
}
```

Even though `a` was listed first, it would be sorted below `b`, because highest index is used when duplicated values are squashed into single key after flipping. I believe Fixer should throw an exception about misconfiguration if there are non-unique attributes listed in `order` option.

Reply:
Done